gsy_framework/sim_results/kpi.py

```diff
@@ -140,22 +147,24 @@ def _dissipate_infinite_consumption(self, trade: Dict):
             self.total_self_consumption_wh += trade["energy"] * 1000
             self.demanded_buffer_wh += trade["energy"] * 1000
 
-    def _accumulate_energy_trace(self, core_stats):
+    def _accumulate_energy_trace(self, core_stats, p2p=True):
         for target_area_uuid in self.areas_to_trace_list:
             target_core_stats = core_stats.get(target_area_uuid, {})
             for trade in target_core_stats.get("trades", []):
                 self._accumulate_self_consumption(trade)
                 self._accumulate_self_production(trade)
                 self._accumulate_self_consumption_buffer(trade)
                 self._dissipate_self_consumption_buffer(trade)
-                self._accumulate_infinite_consumption(trade)
-                self._dissipate_infinite_consumption(trade)
+                if p2p:
+                    self._accumulate_infinite_consumption(trade)
+                    self._dissipate_infinite_consumption(trade)
 
     def update_area_kpi(self, area_dict: Dict, core_stats: Dict):
         """Update kpi after every market cycle"""
         self.total_energy_demanded_wh = 0
+
         self._accumulate_total_energy_demanded(area_dict, core_stats)
-        self._accumulate_energy_trace(core_stats)
+        self._accumulate_energy_trace(core_stats, "P2P" in area_dict["name"])
```

Comment:
Do we really want to add this solution permanently?
If yes, we should add a more unique string IMO.
e.g. `##P2P##` What do you think?

Reply:
Let me think of it a bit more and adapt it. Same goes for the "DH" string for distinguishing district heating.